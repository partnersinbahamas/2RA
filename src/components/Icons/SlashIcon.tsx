import React from 'react';
import PropTypes from 'prop-types';

import { LargeSvg, MediumSvg, SmallSvg } from './Svg';
import { TIcon } from '.';

const CheckIcon: React.FC<TIcon> = ({ className, color, size, style, withBox }) => {
  switch (size) {
    case 'SMALL':
      return (
        <SmallSvg
          xmlns="http://www.w3.org/2000/svg"
          viewBox={withBox ? '-2 -1 28 28' : '0 1 24 24'}
          className={className}
          stroke={color}
          style={style}
        >
          <line x1="2" y1="22" x2="22" y2="4" strokeWidth="4" strokeLinecap="round" />
        </SmallSvg>
      );

    case 'MEDIUM':
      return (
        <MediumSvg
          xmlns="http://www.w3.org/2000/svg"
          viewBox={withBox ? '-3 -3 30 30' : '-1 -1 26 26'}
          className={className}
          stroke={color}
          style={style}
        >
          <line x1="1.5" y1="22.5" x2="22.5" y2="1.5" strokeWidth="3" strokeLinecap="round" />
        </MediumSvg>
      );

    case 'LARGE':
    default:
      return (
        <LargeSvg
          xmlns="http://www.w3.org/2000/svg"
          viewBox={withBox ? '-4 -4 32 32' : '-2 -2 28 28'}
          className={className}
          stroke={color}
          style={style}
        >
          <line x1="1" y1="23" x2="23" y2="1" strokeWidth="2" strokeLinecap="round" />
        </LargeSvg>
      );
  }
};

CheckIcon.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  style: PropTypes.any,
  withBox: PropTypes.bool,
};

CheckIcon.defaultProps = {
  className: '',
  color: 'currentcolor',
  size: 'LARGE',
  style: {},
  withBox: false,
};

export default CheckIcon;
