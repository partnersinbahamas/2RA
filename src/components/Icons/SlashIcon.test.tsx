import { render } from '@testing-library/react';
import React from 'react';

import SlashIcon from './SlashIcon';

describe('SlashIcon', () => {
  it('renders small', () => {
    const { container } = render(<SlashIcon size="SMALL" />);
    expect(container).toMatchSnapshot();
  });
  it('renders medium', () => {
    const { container } = render(<SlashIcon size="MEDIUM" />);
    expect(container).toMatchSnapshot();
  });
  it('renders large', () => {
    const { container } = render(<SlashIcon size="LARGE" />);
    expect(container).toMatchSnapshot();
  });
  it('renders default', () => {
    const { container } = render(<SlashIcon />);
    expect(container).toMatchSnapshot();
  });
});
