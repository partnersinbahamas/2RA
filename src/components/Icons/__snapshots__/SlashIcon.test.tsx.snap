// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlashIcon renders default 1`] = `
<div>
  <svg
    class="sc-hLQSwg hewGWh"
    stroke="currentcolor"
    viewBox="-2 -2 28 28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      stroke-linecap="round"
      stroke-width="2"
      x1="1"
      x2="23"
      y1="23"
      y2="1"
    />
  </svg>
</div>
`;

exports[`SlashIcon renders large 1`] = `
<div>
  <svg
    class="sc-hLQSwg hewGWh"
    stroke="currentcolor"
    viewBox="-2 -2 28 28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      stroke-linecap="round"
      stroke-width="2"
      x1="1"
      x2="23"
      y1="23"
      y2="1"
    />
  </svg>
</div>
`;

exports[`SlashIcon renders medium 1`] = `
<div>
  <svg
    class="sc-dmyCSP dRcSrL"
    stroke="currentcolor"
    viewBox="-1 -1 26 26"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      stroke-linecap="round"
      stroke-width="3"
      x1="1.5"
      x2="22.5"
      y1="22.5"
      y2="1.5"
    />
  </svg>
</div>
`;

exports[`SlashIcon renders small 1`] = `
<div>
  <svg
    class="sc-guDLey kBBjNb"
    stroke="currentcolor"
    viewBox="0 1 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      stroke-linecap="round"
      stroke-width="4"
      x1="2"
      x2="22"
      y1="22"
      y2="4"
    />
  </svg>
</div>
`;
